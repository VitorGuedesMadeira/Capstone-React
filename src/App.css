.main-section {
  background-color: black;
}
