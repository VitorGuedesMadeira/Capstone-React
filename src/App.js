import React from 'react';
import { Route, Routes } from 'react-router-dom';
import Navbar from './Components/Navbar';
import HomePage from './Pages/HomePage';
import ComicsPage from './Pages/ComicsPage';

function App() {
  return (
    <>
      <Navbar />
      <main className="main-section">
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="Comics" element={<ComicsPage />} />
        </Routes>
      </main>
    </>
  );
}

export default App;
