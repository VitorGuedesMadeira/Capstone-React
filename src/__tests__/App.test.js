import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import React from 'react';
import store from '../Redux/configureStore';
import App from '../App';

describe('App js testing', () => {
  test('Renders App', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <App />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
