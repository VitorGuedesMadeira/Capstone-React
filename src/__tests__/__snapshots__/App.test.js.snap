// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App js testing Renders App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navigation-bar"
      >
        <ul
          class="navigation-ul"
        >
          <li>
            <a
              aria-current="page"
              class="home-link active"
              href="/"
            >
              HOME
            </a>
          </li>
        </ul>
        <img
          alt="marvel-logo"
          class="nav-logo"
          src="MarvelStudios.png"
        />
      </nav>
      <main
        class="main-section"
      >
        <div>
          <div
            class="characters-section"
          >
            <div
              class="marvel-characters-title"
            >
              <img
                alt="marvel-heros"
                class="marvel-characters-image"
                src="marvelImage.jpg"
              />
              <div
                class="marvel-characters-text"
              >
                <p
                  class="characters-text"
                >
                  CHARACTERS
                </p>
                <p
                  class="characters-counter"
                >
                  (1562)
                </p>
              </div>
            </div>
            <div
              class="testing"
            >
              <div
                class="marvel-characters"
              >
                <div
                  class="character-title"
                >
                  <div
                    class="search"
                  >
                    <form>
                      <input
                        class="form-control"
                        placeholder="ðŸ”ŽSearch Heros"
                        type="text"
                        value=""
                      />
                    </form>
                  </div>
                </div>
                <div
                  class="loading"
                >
                  Loading...
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navigation-bar"
    >
      <ul
        class="navigation-ul"
      >
        <li>
          <a
            aria-current="page"
            class="home-link active"
            href="/"
          >
            HOME
          </a>
        </li>
      </ul>
      <img
        alt="marvel-logo"
        class="nav-logo"
        src="MarvelStudios.png"
      />
    </nav>
    <main
      class="main-section"
    >
      <div>
        <div
          class="characters-section"
        >
          <div
            class="marvel-characters-title"
          >
            <img
              alt="marvel-heros"
              class="marvel-characters-image"
              src="marvelImage.jpg"
            />
            <div
              class="marvel-characters-text"
            >
              <p
                class="characters-text"
              >
                CHARACTERS
              </p>
              <p
                class="characters-counter"
              >
                (1562)
              </p>
            </div>
          </div>
          <div
            class="testing"
          >
            <div
              class="marvel-characters"
            >
              <div
                class="character-title"
              >
                <div
                  class="search"
                >
                  <form>
                    <input
                      class="form-control"
                      placeholder="ðŸ”ŽSearch Heros"
                      type="text"
                      value=""
                    />
                  </form>
                </div>
              </div>
              <div
                class="loading"
              >
                Loading...
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
